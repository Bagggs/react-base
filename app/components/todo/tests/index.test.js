import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { mount } from 'enzyme';
import { ListGroupItem } from 'react-bootstrap';
import Todo from 'components/todo';

describe('Todo', () => {
  const todo = { id: 1, isComplete: false, name: 'Something to do' };

  it('renders correctly', () => {
    const renderer = new ReactShallowRenderer();

    renderer.render(<Todo todo={ todo }/>);

    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });

  describe('callbacks', () => {
    const updateTodo = jest.fn();
    const deleteTodo = jest.fn();
    const todoComponent = mount(
      <Todo
        updateTodo={ updateTodo }
        deleteTodo={ deleteTodo }
        todo={ todo }
      />
    );

    it('calls toggle() when clicking on li and mark todo as complete', () => {
      todoComponent.find(ListGroupItem).at(0).simulate('click');
      expect(updateTodo).toHaveBeenCalled();
    });

    it('calls delete() when clicking on delete icon', () => {
      todoComponent.find('.glyphicon-trash').simulate('click');
      expect(deleteTodo).toHaveBeenCalled();
    });
  });
});
