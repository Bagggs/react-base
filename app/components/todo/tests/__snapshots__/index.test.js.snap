// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo renders correctly 1`] = `
<ReactCSSTransitionGroup
  transitionAppear={true}
  transitionAppearTimeout={0}
  transitionEnter={true}
  transitionEnterTimeout={0}
  transitionLeave={true}
  transitionLeaveTimeout={0}
  transitionName="todo"
>
  <ListGroupItem
    bsClass="list-group-item"
    listItem={false}
    onClick={[Function]}
  >
    <span>
      Something to do
    </span>
    <span
      className="glyphicon glyphicon-trash trashIcon"
      onClick={[Function]}
    />
  </ListGroupItem>
</ReactCSSTransitionGroup>
`;
